com.unity.shadergraph/Editor/Drawing/PreviewManager.cs

```diff
@@ -799,6 +799,8 @@ void BeginCompile(PreviewRenderData renderData, string shaderStr)
                     ShaderUtil.UpdateShaderAsset(shaderData.shader, shaderStr, false);
                 }
 
+                CoreUtils.Destroy(shaderData.mat);
```

Comment:
@sebastienlagarde Why do we want to destroy the Material here?
Does this magically set shaderData.mat to null?  Otherwise the material might get stuck in a "destroyed but not null" state...

Reply:
@FrancescoC-unity ^^

Reply:
I actually got confused here and I thought it did (our HD util that destroys does set to null, while coreutils. seems not to). 

I am curious on why it did not break though. Visually everything seems to work fine (and QA tested :/ ) 


On the why: it was a workaround that seemed to work as in the keywords were not queried wrongly but the preview material. This is something that was planned to be reverted as soon as the original bug gets fixed.

Reply:
Will check tomorrow morning if adding the = null makes the workaround not work anymore :D

Reply:
Make sure to test it with modifying the upstream graph of a preview node several times.  That should cause this code to be called repeatedly.

Reply:
I tested it locally, seems to work.
I think it's because of this:
https://forum.unity.com/threads/unitys-magic-null-check-not-properly-documented.408135/
"Destroyed" objects are "equal" to null..   :)

Reply:
:D

Ok, to keep it clear I am gonna add the = null assignment anyway, but glad that nothing is broken.